lib/spack/spack/cmd/install.py

```diff
@@ -88,4 +91,6 @@ def install(parser, args):
                 verbose=args.verbose,
                 fake=args.fake,
                 dirty=args.dirty,
-                explicit=True)
+                explicit=True,
+                stop_at=args.stop_at
```

Comment:
I'm not sure how I feel about this design. Currently, Spack supports the following commands:

1. spack fetch
2. spack stage
3. spack patch
4. spack install
5. spack activate

If I try to run stage, it first runs fetch and then stage. If I run install, it does all of the preceding steps. You're adding a new mechanic that does something identical, but with a completely different syntax. Why not do something like this:

1. spack fetch
2. spack stage
3. spack patch
4. spack config/configure/setup
5. spack build
6. spack build-test
7. spack install
8. spack install-test
9. spack activate

Each of these commands would run all of the prior steps if they are implemented and skip them if not.

Obvious downsides:

- There is already a spack config and spack setup. No idea what they do.
- We would have to name the methods similarly for each package subclass.

The former is a serious concern, the latter isn't in my mind. Yes, not all packages will have each of those stages, but we can do nothing and raise a warning in that case.

Some concerns with both my design and yours:

- What will `spack help` list as options?
- What will `spack help install` list as options?
- What will happen when I try to run/stop at a stage that doesn't exist?
- If I run `spack build` and then `spack install`, will it start from the beginning again or pick up where it left off? How will it know that the previous stage succeeded?

Either way, I don't see myself running any of these in-between stages myself. I plan on either running `spack install` or `spack install -t` if I want to run all tests. As long as that works, I don't mind.